server/datastore/mysql/vpp.go

```diff
@@ -253,9 +297,24 @@ func (ds *Datastore) SetTeamVPPApps(ctx context.Context, teamID *uint, appFleets
 
 	for _, appFleet := range appFleets {
 		// upsert it if it does not exist or SelfService or InstallDuringSetup flags are changed
-		if existingFleet, ok := existingApps[appFleet.VPPAppID]; !ok || existingFleet.SelfService != appFleet.SelfService ||
+		existingFleet, ok := existingApps[appFleet.VPPAppID]
```

Comment:
```suggestion
		existingApp, isExistingApp := existingApps[appFleet.VPPAppID]
```

`existingApp` from the previous for shouldn't be in-scope here so we can reuse the name, as it's the same type etc.